library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java

```diff
@@ -106,94 +114,97 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     @Override
-    public void onResult(@NonNull LocationSettingsResult result) {
-        final Status status = result.getStatus();
-
-        switch (status.getStatusCode()) {
-            case LocationSettingsStatusCodes.SUCCESS:
-                // All location settings are satisfied. The client can initialize location
-                // requests here.
-                LogUtils.logI("We got GPS, Wifi and/or Cell network providers enabled enough "
-                      + "to receive location as we needed. Requesting location update...");
-                requestLocationUpdate();
-                break;
-            case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
-                // Location settings are not satisfied.
-                // However, we have no way to fix the settings so we won't show the dialog.
-                LogUtils.logE("Settings change is not available, SettingsApi failing...");
-                settingsApiFail(FailType.GOOGLE_PLAY_SERVICES_SETTINGS_DIALOG);
-                break;
-            case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
-                // Location settings are not satisfied. But could be fixed by showing the user
-                // a dialog.
-                resolveSettingsApi(status);
-                break;
+    public void onResult(@NonNull Task<LocationSettingsResponse> resultTask) {
+
+        try {
+            //noinspection unused
+            LocationSettingsResponse response = resultTask.getResult(ApiException.class);
+            // All location settings are satisfied. The client can initialize location
+            // requests here.
+            LogUtils.logI("We got GPS, Wifi and/or Cell network providers enabled enough "
+                    + "to receive location as we needed. Requesting location update...");
+            requestLocationUpdate();
+        } catch (ApiException exception) {
+            switch (exception.getStatusCode()) {
+                case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+                    mRequestingLocationUpdates = false;
+                    // Location settings are not satisfied.
+                    // However, we have no way to fix the settings so we won't show the dialog.
+                    LogUtils.logE("Settings change is not available, SettingsApi failing...");
+                    settingsApiFail(FailType.GOOGLE_PLAY_SERVICES_SETTINGS_DIALOG);
+
+                    break;
+                case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
+                    // Location settings are not satisfied. But could be fixed by showing the user
+                    // a dialog.
+                    try {
+                        // Cast to a resolvable exception.
+                        resolveSettingsApi((ResolvableApiException) exception);
+
+                        break;
+                    } catch (ClassCastException e) {
+                        // Ignore, should be an impossible error.
```

Comment:
Please do not `try/catch` and ignore it. 
If casting here is safe, then remove try/catch - being aggressive here would make more sense. 

On the other hand, can we not call `resultTask.getResult(ResolvableApiException.class)` directly?

Reply:
Split listener to onSuccess/onFailure, remove try/catch.